### PR TITLE
🔧 bsx/snek offers composition api

### PR DESCRIPTION
**Pull request by roiLeo**
- part of #4750 
- & #6392 

Screenshot 2023-08-09 at 13-50-12 KodaDot - NFT Market Explorer

note: I didn't test all features (looks like doesn't work on beta)

### CODE REVIEW

**Comment by prury** [comment]
luckily i have some offers for testing purposes:

image

image

image

File: components/bsx/Offer/MyOffer.vue [hunk 1]
```diff
@@ -85,99 +82,155 @@
...
+onMounted(() => {
+  if (targetAddress.value) {
+    // $apollo.addSmartQuery<OfferResponse>('offers', {
+    //   client: client.value,
+    //   query: acceptableOfferByCurrentOwner,
+    //   variables: { id: targetAddress.value },
+    //   manual: true,
+    //   result: ({ data }) => setResponse(data),
+    //   pollInterval: 10000,
+    // })
```
Comment (vikiival) [inline components/bsx/Offer/MyOffer.vue:161]: how do you manage refreshing?

**Review by vikiival — COMMENTED** [review]
Otherwise LGTM

**Comment by roiLeo** [comment]
> luckily i have some offers for testing purposes:

Should be fixed now (I guess??)

Screenshot 2023-08-09 at 16-04-37 KodaDot - NFT Market Explorer

**Comment by prury** [comment]
> > luckily i have some offers for testing purposes:
> 
> Should be fixed now (I guess??)
> 
> Screenshot 2023-08-09 at 16-04-37 KodaDot - NFT Market Explorer

yeee

image

File: components/bsx/Offer/MyOffer.vue (hunk 1 above)
Comment (vikiival) [reply components/bsx/Offer/MyOffer.vue]: GalleryItem should be a separate data island

Comment (roiLeo) [reply components/bsx/Offer/MyOffer.vue]: Introducing manual refresh button

Screenshot 2023-08-10 at 10-24-08 KodaDot - NFT Market Explorer

**Pull request merged**